resolver/resolver.go

```diff
@@ -467,29 +472,28 @@ func truncate(m *dns.Msg, udp bool) *dns.Msg {
 
 	furtherTruncation := m.Len() > max
 	m.Truncated = m.Truncated || furtherTruncation
-
+	if !setTruncateBit {
+		// Clear the Truncate bit regardless of whether this message used
+		// to have it set set or whether we are going to truncate it here.
+		m.Truncated = false
+	}
 	if !furtherTruncation {
 		return m
 	}
-
 	m.Extra = nil // Drop all extra records first
 	if m.Len() < max {
+		// Dropping the extra records shrunk the message down sufficiently.
```

Comment:
I don't believe so - shrunk is the past participle of shrink (thanks google.) I shrink, yesterday I shrunk. I'll reword the comment anyway.